go/p2p/rpc/rpc.go

```diff
@@ -23,6 +23,11 @@ type P2P interface {
 	GetHost() core.Host
 }
 
+// NewProtocolID generates a protocol identifier for a consensus P2P protocol.
+func NewProtocolID(protocolID string, version version.Version) protocol.ID {
+	return protocol.ID(fmt.Sprintf("/oasis/%s/consensus/%s", protocolID, version.MaskNonMajor()))
```

Comment:
Runtime protocol ID is:
`/oasis/[protocol_id]/[runtime_id]/[version]`

The currently proposed consensus protocol ID is:
`/oasis/[protocol_id]/consensus/[version]`

---

This also made me think, should we include chain context in the protocol identifiers? (also for runtime protocols)

Reply:
It would be nice if mainnet and testnet protocol IDs would differ so that we can use the same discovery techniques for both of them (seed node, pex, ..). Otherwise, we will have to prefix them somewhere, e.g. in a discovery wrapper. 

The protocol ID should be unique, so no need for `consensus` part.

Topics are named `[chain_context]/[comittee_version]/[runtime_id]/[kind]` and `pubsub` prefixes this with `floodsub:` to get the protocol ID or namespace for that topic. I think that in the next release we can change the names, so maybe use something like this:
- Runtime: `/oasis/[chain_context]/[protocol_id]/[runtime_id]/[version]` 
- Consensus:  `/oasis/[chain_context]/[protocol_id]/[version]`
- Topic: `/oasis/[chain_context]/[kind]/[runtime_id]/[comittee_version]`
Topic kinds and protocol IDs should be unique.

Reply:
Yeah we can afford protocol-breaking changes in master, so we can change these to include the chain context. The above scheme makes sense, I'll open a task so we don't forget about it.

Reply:
Ok, i'll drop the "consensus" part for now, and we can update to include chain-context in a followup PR.